src/commands/search.ts

```diff
@@ -81,24 +54,17 @@ export class SearchCommand extends Command {
 	}
 
 	protected override async execute(interaction: ChatInputCommandInteraction): Promise<number> {
-		const type = interaction.options.getSubcommand(true) as CardLookupType;
-		const input = interaction.options.getString("input", true);
-		const resultLanguage = await this.locales.get(interaction);
-		const inputLanguage = (interaction.options.getString("input-language") as Locale) ?? resultLanguage;
-		// Send out both requests simultaneously
-		const [, card] = await Promise.all([interaction.deferReply(), getCard(this.got, type, input, inputLanguage)]);
-		let end: number;
+		const { type, input, resultLanguage, inputLanguage } = await getCardSearchOptions(interaction, this.locales);
+		const card = await getCard(this.got, type, input, inputLanguage);
+		let replyOptions;
 		if (!card) {
-			end = Date.now();
 			useLocale(resultLanguage);
-			await interaction.editReply({ content: t`Could not find a card matching \`${input}\`!` });
+			replyOptions = { content: t`Could not find a card matching \`${input}\`!` };
 		} else {
 			const embeds = createCardEmbed(card, resultLanguage);
-			end = Date.now();
-			await interaction.editReply({ embeds }); // Actually returns void
+			replyOptions = { embeds };
 		}
-		// When using deferReply, editedTimestamp is null, as if the reply was never edited, so provide a best estimate
-		const latency = end - interaction.createdTimestamp;
-		return latency;
+		const reply = await interaction.reply({ ...replyOptions, fetchReply: true });
```

Comment:
Right, we don't defer this one, because all we need to do is get the card.

Reply:
/art and /price are deferred on the success path because we don't have an SLA for the art server or the price API. Search though is basically guaranteed to respond in under 200 ms. The full timeouts we've been seeing are rare, bizarre aberrations to investigate.